pkg/util/merge/merge_automationconfigs_test.go

```diff
@@ -0,0 +1,38 @@
+package merge
+
+import (
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/automationconfig"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestMergeAutomationConfigs(t *testing.T) {
```

Comment:
I think we could add a test where the override changes a process that does not exist in the original automation config. This can be achieved by an automation config override with a different name, or with more elements, where the ones with modifications are over the len() of the original automationconfig.